vignettes/design.qmd

```diff
@@ -58,4 +58,159 @@ if the processing time per file is substantial.
+#' @param output_path The path with the directory to save the output Parquet file to.
+#'
+#' @returns Returns a character vector with the path to the created Parquet file(s)
+#'   (from `output_path`), so it can easily be used in a targets pipeline.
+#' @export
+#' @examples
+#' \dontrun{
+#' convert_register_to_parquet(c("bef2018.sas7bdat",
+#'      "bef2019.sas7bdat",
+#'      "Demember2023bef2019.sas7bdat"), "output_directory/")
+#' }
+convert_register_to_parquet <- function(input_path, output_path) {
+  # Initial setup and checks.
+  fs::file_exists(input_path)
+  checkmate::assert_character(input_path)
+  checkmate::assert_character(output_path)
+  checkmate::assert_scalar(output_path)
+  fs::dir_create(fs::path_dir(output_path))
+
+  # Read SAS file(s) and full join.
+  data <- read_sas_files(input_path) |>
+    # Add year column if possible.
+    add_year_col() |>
+    # Deduplicate identical columns.
+    dplyr::distinct() |>
+    # TODO: How to handle history/almost-duplicate rows?
+    # Write to Parquet.
+    # TODO: Might have to do some processing of the dates. If so, create a helper function for this.
+    # mutate(across(where(~inherits(.x, what = "date")), as.character))
+    # mutate(across(where(lubridate::is.Date), as.character))
+    arrow::write_parquet(output_path)
+    # TODO: How is the year partitioning handled in/passed to arrow::write_parquet()?
+
+
+  if (length(input_path) > 1) {
+    cli::cli_alert_success(
+      "Finished merging and saving the duplicate {.val {fs::path_file(input_path)[1]}} files as a Parquet file."
+    )
+  } else {
+    cli::cli_alert_success(
+      "Finished saving {.val {fs::path_file(input_path)}} as a Parquet file."
+    )
+  }
+
+  output_path
+}
+
+
+
+read_sas_files <- function(input_path) {
+  input_path |>
+    purrr::map(haven::read_sas) |>
+    purrr::reduce(dplyr::full_join)
+}
+
+add_year_col <- function(data) {
+  year <- get_database_year(input_path) |>
+    unique()
+  if (!is.na(year) & year %in% 1969:2030) {
+    data <- data |>
+      dplyr::mutate(year = year)
+  } else {
+    cli::cli_alert_info(
+      "Could not determine year from file name(s). Will not partition by year."
+    )
+  }
+  data
+}
+```
+
+### Helper functions
+
+<!-- TODO: Do we want to include the helper functions for duplicate and non-duplicate paths? -->
+
+#### `list_files_with_year(path)`
+
+Lists all files in a given directory that contain a year (four
+consecutive digits) in their filenames. It returns a character vector of
+file paths that match this criterion.
+
+<!-- TODO: Add link to reference documentation when ready -->
+
+#### `list_files_without_year(path)`
+
+Lists all files in a given directory that **do not** contain a year
+(four consecutive digits) in their filenames. It returns a character
+vector of file paths that match this criterion.
+
+<!-- TODO: Add link to reference documentation when ready -->
+
+#### `list_register_files(path, register_name)`

```

Comment:
Hmm, I'm not sure this function is necessary, at least not as an external function.